flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java

```diff
@@ -268,6 +277,26 @@ private void expand() {
                                     + " has set uid for some operators.");
                 }
 
+                // Set the operator uid hashes to support stateful upgrades without prior uids
+                if (hasWriterUidHash) {
+                    if (subTransformation.getName().equals(WRITER_NAME)) {
```

Comment:
Is this safe?

Reply:
I guess this can cause problems if other transformations use the same name. For the Writer and Committer there might be another way to set the properties but I currently do not see a way to change it for the GlobalCommitter because it is added as part of a post commit topology and the content of the topology is not accessible from the translator so I have to rely on the naming either way.

Any ideas?

Reply:
I talked to @AHeise about this problem offline and we did not find a ideal solution even if only considering the writer and committer. The only idea was to pull all writer and committer configuration up to the point where we add the writer/committer to the datastream and directly manipulate the transformation. Unfortunately, this still implies matching the transformation while looping through the newly added transformation because we now would need to skip the already configured transformation.

Overall, we evaluated the problem again and only occurs as a problem iff some of the sub topology (pre-writer, pre/post committer) includes a transformation matching the name of either the writer, committer or global committer. We do not expect that this problem is likely to occur very often.